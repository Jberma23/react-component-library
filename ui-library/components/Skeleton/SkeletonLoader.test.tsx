import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { SkeletonLoader } from "./index";

describe(SkeletonLoader, () => {
  it("renders with defaults", () => {
    const { asFragment } = render(<SkeletonLoader />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("appends classNames correctly", () => {
    render(
      <SkeletonLoader
        containerTestId="addl-classnames"
        className="my-test-class"
      />
    );
    const skeletonClass = screen
      .getByTestId("addl-classnames")
      .children[0].getAttribute("class");
    expect(skeletonClass).toContain("my-test-class");
  });

  it("applies animation preference correctly", () => {
    render(
      <SkeletonLoader
        containerTestId="addl-classnames"
        enableAnimation={false}
      />
    );
    const skeletonClass = screen
      .getByTestId("addl-classnames")
      .children[0].getAttribute("class");
    expect(skeletonClass).not.toContain("motion-safe:animate-pulse");
  });
});
