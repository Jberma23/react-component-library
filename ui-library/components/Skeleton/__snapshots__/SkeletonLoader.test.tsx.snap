// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonLoader renders with defaults 1`] = `
<DocumentFragment>
  <span
    aria-busy="false"
    aria-live="polite"
  >
    <span
      class="react-loading-skeleton motion-safe:animate-pulse"
      style="--pseudo-element-display: none; border-radius: 3px; --base-color: #E5E5E5;"
    >
      â€Œ
    </span>
    <br />
  </span>
</DocumentFragment>
`;
